Vagrantfile

```diff
@@ -480,7 +480,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
 
   # Docker use image.
   config.vm.provider :docker do |d, override|
-    d.image = 'pentatonicfunk/vagrant-ubuntu-base-images:20.04'
+    d.image = 'pentatonicfunk/vagrant-ubuntu-base-images:24.04'
```

Comment:
does this trigger a VM rebuild for exsting Ubuntu 22 instances that upgrade?

Reply:
good question, i dont know. it would be vagrant territory, no ? the decision whether docker container need rebuild or not would be based on what vagrant flow will be issued. in any case better to test it, i'll try to test

Reply:
main concerns is if vagrant would kick up a fuss about mismatching images or if it would discard the container ( and the database + data ) to create a new 24 container

Reply:
so i tested this, this is how i tested it

### use `upstream/develop` branch
- Vagrantfile still using 20.04 image
- do `vagrant up --provision`
- verify ubuntu version via `vagarant ssh`
```shell
vagrant@vvv:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.6 LTS
Release:        20.04
Codename:       focal
```
- create new post in one.wordpress.test to verify DB okay
<img width="1052" alt="image" src="https://github.com/user-attachments/assets/21fbb8ec-04b2-49a7-93e1-f174ea580d18">

### switch to `origin/docker-provider`
- Vagrantfile now 24.4
-  do `vagrant halt` to stop docker
- do `vagrant reload --provision` to trigger reprovision
- verify ubuntu version via `vagarant ssh`, expecting the Ubuntu should stay in 20.04
```shell
Last login: Fri Nov 15 15:33:51 2024 from 172.17.0.1
vagrant@vvv:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.6 LTS
Release:        20.04
Codename:       focal
```
- verify whether DB not discarded ( post still exists )
<img width="1565" alt="image" src="https://github.com/user-attachments/assets/ae288508-a98e-429f-94be-58e8fa1b4d83">
 
 i think this is what we expected, right ?
- no complains on provisioner
- container not discarded
- DB not discarded

i think what we can do is, add somewhere in docs ( if we haven't), if user wants to upgrade their OS is by doing 
- `db_backup` 
- `vagrant destroy`
- `vagrant up --provision` --> this step will trigger 24.04
- `db_restore`